src/Joystick/Joystick.cc

```diff
@@ -687,6 +689,19 @@ void Joystick::setThrottleMode(int mode)
     emit throttleModeChanged(_throttleMode);
 }
 
+bool Joystick::negativeThrust(void)
+{
+    return _negativeThrust;
+}
+
+void Joystick::setNegativeThrust(bool allowNegative)
+{
+    _negativeThrust = allowNegative;
```

Comment:
There should be a check here whether allowNegative is actually different than the current setting. And only in that case do you need to do anything here. That prevent unnecessary signaling.